src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/mailActionFields.tsx

```diff
@@ -0,0 +1,123 @@
+    /**
+     * TargetIdentifiers between the targetTypes are not unique, and may wrongly map to something that has not been
+     * selected. E.g. A member and project can both have the `targetIdentifier`, `'2'`. Hence we clear the identifier.
+     **/
+    if (attribute === 'targetType') {
+      newAction.targetIdentifier = '';
+    }
+    onChange(newAction);
+    /**
+     * Force update is needed because parent component mutates its props with the new values that have changed.
+     * Effectively, this causes `action`, a nested prop from the parent, to be referentially the same even if a new
+     * `action` is passed to the change handler.
+     **/
+    this.forceUpdate();
+  };
+
+  handleChangeActorType = (e: HTMLInputElement) => {
+    this.handleChange('targetType', e);
+  };
+
+  handleChangeActorId = (e: HTMLInputElement) => {
+    this.handleChange('targetIdentifier', e);
+  };
+
+  render: () => React.ReactElement = () => {
+    const {disabled, loading, projects, project, organization, action} = this.props;
+    const {slug: projectSlug} = project;
+
+    const isIssueOwners = action.targetType === MailActionTargetType.IssueOwners;
+    const isTeam = action.targetType === MailActionTargetType.Team;
+
+    return (
+      <PanelItemGrid>
+        <SelectControl
+          isClearable={false}
+          disabled={disabled || loading}
+          value={action.targetType}
+          styles={{
+            control: provided => ({
+              ...provided,
+              minHeight: '28px',
+              height: '28px',
+            }),
+          }}
+          options={[
+            {value: MailActionTargetType.IssueOwners, label: 'Issue Owners'},
+            {value: MailActionTargetType.Team, label: 'Team'},
+            {value: MailActionTargetType.Member, label: 'Member'},
+          ]}
+          onChange={this.handleChangeActorType}
+        />
+        {!isIssueOwners ? (
+          <SelectMembers
+            disabled={disabled}
+            key={isTeam ? MailActionTargetType.Team : MailActionTargetType.Member}
+            showTeam={isTeam}
+            project={projects.find(({slug}) => slug === projectSlug)}
+            organization={organization}
+            // The value from the endpoint is of type `number`, `SelectMembers` require value to be of type `string`
+            value={`${action.targetIdentifier}`}
+            styles={{
+              control: provided => ({
+                ...provided,
+                minHeight: '28px',
+                height: '28px',
+              }),
+            }}
+            onChange={this.handleChangeActorId}
+          />
+        ) : (
+          <span />
+        )}
+      </PanelItemGrid>
+    );
+  };
+}
+
+const PanelItemGrid = styled(PanelItem)`
+  display: grid;
+  grid-template-columns: 200px 200px;
+  padding: 0;
+  align-items: center;
+  grid-gap: ${space(2)};
+`;
+
+export default withOrganization(withProjects(withProject(MailActionFields)));
```

Comment:
Let's pass down `org` and `project` from `RuleNode`

Reply:
Thanks! Done.